be/src/exec/parquet/column_reader.h

```diff
@@ -27,6 +30,269 @@ struct ColumnReaderOptions {
+    using ColumnType = typename vectorized::RunTimeTypeTraits<dst_type>::ColumnType;
+
+    BinaryToDecimalConverter(int32_t src_scale, int32_t dst_scale) {
+        if (src_scale < dst_scale) {
+            _scale_type = SCALE_UP;
+            _scale_factor = get_scale_factor<DestPrimitiveType>(dst_scale - src_scale);
+        } else if (src_scale > dst_scale) {
+            _scale_type = SCALE_DOWN;
+            _scale_factor = get_scale_factor<DestPrimitiveType>(src_scale - dst_scale);
+        }
+    }
+
+    Status convert(const vectorized::ColumnPtr& src, vectorized::Column* dst) override {
+        auto* src_nullable_column = vectorized::ColumnHelper::as_raw_column<vectorized::NullableColumn>(src);
+        // hive only support null column
+        // TODO: support not null
+        auto* dst_nullable_column = down_cast<vectorized::NullableColumn*>(dst);
+        dst_nullable_column->resize(src_nullable_column->size());
+
+        auto* src_column =
+                vectorized::ColumnHelper::as_raw_column<vectorized::BinaryColumn>(src_nullable_column->data_column());
+        auto* dst_column = vectorized::ColumnHelper::as_raw_column<ColumnType>(dst_nullable_column->data_column());
+
+        auto& src_data = src_column->get_data();
+        auto& dst_data = dst_column->get_data();
+        auto& src_null_data = src_nullable_column->null_column()->get_data();
+        auto& dst_null_data = dst_nullable_column->null_column()->get_data();
+
+        size_t size = src_column->size();
+        bool has_null = false;
+        for (size_t i = 0; i < size; i++) {
+            // If src_data[i].size > DestPrimitiveType, this means  we treat as null.
+            dst_null_data[i] = src_null_data[i] | (src_data[i].size > sizeof(DestPrimitiveType));
+            if (dst_null_data[i]) {
+                has_null = true;
+                continue;
+            }
+
+            // When Decimal in parquet is stored in byte arrays, binary and fixed,
+            // the unscaled number must be encoded as two's complement using big-endian byte order.
+            DestPrimitiveType value = src_data[i].data[0] & 0x80 ? -1 : 0;
+            memcpy(reinterpret_cast<char*>(&value) + sizeof(DestPrimitiveType) - src_data[i].size, src_data[i].data,
+                   src_data[i].size);
+            value = BitUtil::big_endian_to_host(value);
+
+            // scale up/down
+            if (_scale_type == SCALE_UP) {
+                value *= _scale_factor;
+            } else if (_scale_type == SCALE_DOWN) {
+                value /= _scale_factor;
+            }
+            dst_data[i] = DecimalType(value);
+        }
+        dst_nullable_column->set_has_null(has_null);
+        return Status::OK();
+    }
+
+private:
+    DecimalScaleType _scale_type = NO_SCALE;
+    DestPrimitiveType _scale_factor = 1;
+};
+
+class ColumnConverterFactory {
+public:
+    static Status init_converter(const ParquetField* field, const TypeDescriptor& typeDescriptor,
```

Comment:
const ParquetField& instead of const ParquetField*

Reply:
done